### PR TITLE
unique_ptr homework

### PR DESCRIPTION
**Pull request by jakjan95**
Author @jakjan95

### CODE REVIEW

File: homework/unique_ptr/include/UniquePtr.hpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    delete ptr_;
+    ptr_ = ptr.ptr_;
+    ptr.ptr_ = nullptr;
+    return *this;
+}
+
+template <typename T>
+T UniquePtr<T>::operator*() const {
+    if (!ptr_) {
+        throw NullptrDereferenceError();
```
Comment (MaciejSalamonski) [inline homework/unique_ptr/include/UniquePtr.hpp:56]: +1 for throwing an exception

**Review by MaciejSalamonski — APPROVED** [review]
Very nice code @jakjan95  :)

File: homework/unique_ptr/tests/UniquePtr.ut.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+TEST_F(MoveTest, MoveConstructorTest) {
+    UniquePtr<int> ptr{moveTest()};
+    ASSERT_EQ(9, *ptr);
+}
+
+TEST_F(MoveTest, MoveAssignmentTest) {
+    UniquePtr<int> ptr{new int{12}};
+    ptr = moveTest();
+    ASSERT_EQ(9, *ptr);
+}
```
Comment (ziobron) [inline homework/unique_ptr/tests/UniquePtr.ut.cpp:49]: you should check what happens to source pointers in move operations

**Review by ziobron — CHANGES_REQUESTED** [review]
There is no test for arrow operator.

**Comment by jakjan95** [comment]
Tests added, stuff mentioned by Lukasz fixed :)

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
{ "requirement_ids": [3784, 1119] }
36 XP za implementację unique_ptra (3784)
3 XP za punktualność

**Pull request closed without merging**